### ISSUE TITLE
Support progressive append/udpate (same variable multiple times in env file)

### ISSUE THREAD

**Issue by kanaka** [dotenvx/dotenvx#441]
This is a followup to https://github.com/dotenvx/dotenvx/issues/433

We have several different contexts where we generate a `.env` file by combining multiple other files (and sometimes adding or transforming them along the way).

For example, consider three env files:
```
# config1 definitions
options="${options} optA"
configX=blah
```
```
# config2 definitions
options="${options} optB optC ${configX:+optX}"
```
```
# config3 defintions
options="${options} optD"
```

Those files get combined into a single `.env` in a specific order (based on direct and transitive dependencies). The resulting file is:
```
# config1 definitions
options="${options} optA"
configX=blah
# config2 definitions
options="${options} optB optC ${configX:+optX}"
# config3 defintions
options="${options} optD"
```

The value of resulting loading vars should be:
```
configX=blah
options=" optA optB optC optX optD"
```

That's the behavior is you source the file with bash and also the behavior if you use the .env file in a docker compose context.

Here is a simple docker-compose file that will load an env file and print the environment (minus some bash stuff):
```
services:
  foo:
    image: ubuntu
    env_file: .env
    command: bash -c 'env | grep "^[A-Z]" | egrep -v "HOME=|PWD|SHLVL|TERM" | sort'
```

Here is a bash command that can be use to load an env-file without any extraneous values from the environment:
```
bash -c '_AAAAA=$(declare -x | sort); set -a; source .env; _ZZZZZ=$(declare -x | sort); comm -13 <(echo "${_AAAAA}") <(echo "${_ZZZZZ}")  | sed "s/^declare -x //"'
```
The bash wrapper is a neat hack, but it has the major downside that you have to completely trust the content of the `.env` because you are essentially executing it raw (and it also depends on the commands: bash, sort, and comm).

It would be great to have a env file loading option outside of docker compose but that still maintained all the useful bash/compose variable expansion capability.

Given the current use of dotenv to load the full env file prior to doing parameter expansions, I think this capability would imply a refactoring so that expansion can happen during env file parsing (after each line). So certainly not trivial to add/fix this.

**Comment by motdotla** [comment]
effort started here: https://github.com/dotenvx/dotenvx/pull/445

**Comment by motdotla** [comment]
https://github.com/dotenvx/dotenvx/pull/445

released as `1.24.0`
